test/qr.jl

```diff
@@ -1,223 +1,62 @@
 using MatrixAlgebraKit
 using Test
-using TestExtras
 using StableRNGs
 using LinearAlgebra: diag, I, Diagonal
+using CUDA, AMDGPU, GenericLinearAlgebra
 
 BLASFloats = (Float32, Float64, ComplexF32, ComplexF64)
 GenericFloats = (Float16, BigFloat, Complex{BigFloat})
 
-@testset "qr_compact! and qr_null! for T = $T" for T in BLASFloats
-    rng = StableRNG(123)
-    m = 54
-    for n in (37, m, 63)
-        minmn = min(m, n)
-        A = randn(rng, T, m, n)
-        Q, R = @constinferred qr_compact(A)
-        @test Q isa Matrix{T} && size(Q) == (m, minmn)
-        @test R isa Matrix{T} && size(R) == (minmn, n)
-        @test Q * R ≈ A
-        N = @constinferred qr_null(A)
-        @test N isa Matrix{T} && size(N) == (m, m - minmn)
-        @test isisometric(Q)
-        @test maximum(abs, A' * N) < eps(real(T))^(2 / 3)
-        @test isisometric(N)
-
-        Ac = similar(A)
-        Q2, R2 = @constinferred qr_compact!(copy!(Ac, A), (Q, R))
-        @test Q2 === Q
-        @test R2 === R
-        N2 = @constinferred qr_null!(copy!(Ac, A), N)
-        @test N2 === N
-
-        Q2 = similar(Q)
-        noR = similar(A, minmn, 0)
-        qr_compact!(copy!(Ac, A), (Q2, noR))
-        @test Q == Q2
-
-        # unblocked algorithm
-        qr_compact!(copy!(Ac, A), (Q, R); blocksize = 1)
-        @test Q * R ≈ A
-        @test isisometric(Q)
-        qr_compact!(copy!(Ac, A), (Q2, noR); blocksize = 1)
-        @test Q == Q2
-        qr_compact!(copy!(Ac, A), (Q2, noR); blocksize = 1)
-        qr_null!(copy!(Ac, A), N; blocksize = 1)
-        @test maximum(abs, A' * N) < eps(real(T))^(2 / 3)
-        @test isisometric(N)
-        if n <= m
-            qr_compact!(copy!(Q2, A), (Q2, noR); blocksize = 1) # in-place Q
-            @test Q ≈ Q2
-            @test_throws ArgumentError qr_compact!(copy!(Q2, A), (Q2, R); blocksize = 1)
-            @test_throws ArgumentError qr_compact!(copy!(Q2, A), (Q2, noR); positive = true)
-            @test_throws ArgumentError qr_compact!(copy!(Q2, A), (Q2, noR); blocksize = 8)
+@isdefined(TestSuite) || include("testsuite/TestSuite.jl")
+using .TestSuite
+
+is_buildkite = get(ENV, "BUILDKITE", "false") == "true"
+
+m = 54
+for T in (BLASFloats..., GenericFloats...), n in (37, m, 63)
+    TestSuite.seed_rng!(123)
+    if is_buildkite && T ∈ BLASFloats
```

Comment:
Can we slightly alter the logic for this, such that if we would locally trigger the tests, the GPU tests also run? I.e. probably its better to exclude things if `!is_buildkite`

Reply:
OK, done, I think